microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks.md

```diff
@@ -0,0 +1,120 @@
+---
+description: Use the Detached Elements tool in the Microsoft Edge DevTools to find and fix DOM memory leaks.
+title: Debug DOM memory leaks
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.date: 11/29/2021
+ms.topic: article
+ms.prod: microsoft-edge
+keywords: microsoft edge, web development, f12 tools, devtools, detached elements, document object model, dom, memory, leak, node
+---
+
+# Debug DOM memory leaks with the Detached Elements tool
+
+A memory leak can occur in your application when an element is no longer attached to the Document Object Model (DOM) tree, but is still referenced by some JavaScript running on the page. These elements are called *detached elements*. For the browser to garbage collect (GC) the detached element, the element must not be referenced from either the DOM tree or JavaScript code.
+
+This article describes how to use the **Detached Elements** tool in Microsoft Edge DevTools to find detached elements that the browser cannot GC in a demo application and identify the JavaScript that is still referencing the detached element. By changing your JavaScript to release the element, you reduce the number of detached elements on your page.
+
+For more information about heap snapshots and detached elements, read [Discover detached DOM tree memory leaks with Heap Snapshots](/devtools-guide-chromium/memory-problems/#discover-detached-dom-tree-memory-leaks-with-heap-snapshots).
+
+## Open the Detached Elements tool
+
+The **Detached Elements** tool is available as an experiment beginning with Microsoft Edge version 93 and is available by default in Microsoft Edge version 97 and later. Check your version of Microsoft Edge by navigating to `edge://version`. If your version is less than 97, read [Turning an experiment on or off](/devtools-guide-chromium/experimental-features/#turning-an-experiment-on-or-off) for instructions about turning on the **Detached Elements** experiment.
+
+To open the **Detached Elements** tool for the demo application:
+
+1. Open the [demo application](https://microsoftedge.github.io/Demos/detached-elements/).
+
+1. [Open DevTools](/devtools-guide-chromium/#opening-devtools).
+
+1. Click the **+** button in the toolbar.
+
+1. Click **Detached Elements**.
+
+:::image type="complex" source="./images/open-detached-elements.msft.png" alt-text="Open the Detached Elements tool." lightbox="./images/open-detached-elements.msft.png":::
+   Open the Detached Elements tool
+:::image-end:::
+
+## Get Detached Elements
```

Comment:
Headers don't need to be capitalized, this can be changed to `Get detached elements` (same for the other headers).